### PR TITLE
chore: migrate typed-enum function

### PR DESCRIPTION
**Pull request by P0lip**
Closes https://github.com/stoplightio/spectral/issues/924

**Checklist**

- [x] Tests added / updated
- [x] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/rulesets/oas/__tests__/typed-enum.ts [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
+import { functions } from '../../../functions';
 import { RuleType, Spectral } from '../../../index';
+import { setFunctionContext } from '../../evaluators';
+import { typedEnum } from '../functions/typedEnum';
 import { rules } from '../index.json';
 
 describe('typed-enum', () => {
   const s = new Spectral();
-  s.setFunctions({ typedEnum });
+  s.setFunctions({ typedEnum: setFunctionContext({ functions }, typedEnum) });
```
Comment (nulltoken) [inline src/rulesets/oas/__tests__/typed-enum.ts:10]: @P0lip re `setFunctionContext` , none of the other custom function test files leverage it. When should it be used?

File: src/rulesets/oas/functions/__tests__/typedEnum.test.ts [hunk 2]
```diff
@@ -1,9 +1,11 @@
+import { functions } from '../../../../functions';
 import { typedEnum } from '../typedEnum';
 
 const defaultReportingThreshold = 3;
 
 function runTypedEnum(targetVal: any, reportingThreshold: any) {
-  return typedEnum(
+  return typedEnum.call(
```
Comment (nulltoken) [inline src/rulesets/oas/functions/__tests__/typedEnum.test.ts:7]: @P0lip Same question here. When should this invocation method be favored?

File: src/rulesets/oas/__tests__/typed-enum.ts (hunk 1 above)
Comment (P0lip) [reply src/rulesets/oas/__tests__/typed-enum.ts:10]: This is how we inject these built-in functions.
It's needed to use `setFunctionContext` if your custom functions makes use of the built-in one.
Spectral takes care of it [here](https://github.com/stoplightio/spectral/blob/develop/src/spectral.ts#L176), but obviously if you have a unit test, you need to do it on your own.

File: src/rulesets/oas/functions/__tests__/typedEnum.test.ts (hunk 2 above)
Comment (P0lip) [reply src/rulesets/oas/functions/__tests__/typedEnum.test.ts:7]: Everything `setFunctionContext` does is setting the "`this` context".
It uses `Function.prototype.bind` underneath.
That being said, you can either use `setFunctionContext` or Function.prototype.{bind,call,apply}, whatever works for you the best.

**Pull request merged**